typescript/lib/networks/network_context.ts

```diff
@@ -0,0 +1,604 @@
+                } else {
+                    resolve(errAsync(error));
+                }
+            })
+        });
+        const resultRegisterStaticFiles: Result<RegisterStaticFilesResponse, Error> = await promiseRegisterStaticFiles;
+        if (!resultRegisterStaticFiles.isOk()) {
+            return err(resultRegisterStaticFiles.error);
+        }
+        const resp: RegisterStaticFilesResponse = resultRegisterStaticFiles.value;
+
+        const staticFileDestRelativeFilepathsMap: Map<string, string> = resp.getStaticFileDestRelativeFilepathsMap();
+        for (let [staticFileIdStr, destFilepathRelativeToEnclaveVolRoot] of staticFileDestRelativeFilepathsMap.entries()) {
+
+            const staticFileId: StaticFileID = <StaticFileID>(staticFileIdStr);
+
+            if (!staticFileFilepaths.has(staticFileId)) {
+                return err(new Error("No source filepath found for static file " + staticFileId + "; this is a bug in Kurtosis"));
+            }
+            const srcAbsFilepath: string = staticFileFilepaths[staticFileId];
+
+            const destAbsFilepath: string = path.join(this.enclaveDataVolMountpoint, destFilepathRelativeToEnclaveVolRoot);
+            
+            const promiseStatDestAbsFilepath: Promise<ResultAsync<fs.Stats, Error>> = new Promise((resolve, _unusedReject) => {
+                fs.stat(destAbsFilepath, (error: Error, response: fs.Stats) => {
+                    if (error === null) {
+                        resolve(okAsync(response));
+                    } else {
+                        resolve(errAsync(error));
+                    }
+                })
+            });
+            const resultStatDestAbsFilepath: Result<fs.Stats, Error> = await promiseStatDestAbsFilepath;
+            if (!resultStatDestAbsFilepath.isOk()) {
+                return err(new Error("The Kurtosis API asked us to copy static file " + staticFileId + " to path " + destFilepathRelativeToEnclaveVolRoot + 
+                " in the enclave volume which means that an empty file should exist there, " + "but no file exists at that path - this is a bug in Kurtosis!"));
+            }
+
+            
+            var srcFp: number;
+            var destFp: number;
+            try {
+                
+                const promiseOpenSrcFp: Promise<ResultAsync<number, Error>> = new Promise((resolve, _unusedReject) => {
+                    fs.open(srcAbsFilepath, 'r', (error: Error, response: number) => {
+                        if (error === null) {
+                            resolve(okAsync(response));
+                        } else {
+                            resolve(errAsync(error));
+                        }
+                    })
+                });
+                const resultOpenSrcFp: Result<number, Error> = await promiseOpenSrcFp;
+                if (!resultOpenSrcFp.isOk()) {
+                    return err(resultOpenSrcFp.error);
+                }
+                srcFp = resultOpenSrcFp.value;
```

Comment:
You've still got the same problem with your `try/catch` - if any code before this line throws an error, you'll call `srcFp.close()` on a null pointer. You should actually put the first `try` after this line (which is why you don't need to pre-declare srcFp or destFp)